### ISSUE TITLE
Yale Door Locks

### ISSUE THREAD

**Issue by simonrb2000** [tonesto7/homebridge-smartthings#67]
Not bought one yet but was hoping to get a Yale Keyfree Smart Lock to work with Homebridge. If I buy a Smartthings hub and use this addon would I be ale to say "unlock front door" or ask "is the front door is lock?"

Cheers guys/girls 

http://www.yale.co.uk/en/yale/couk/smart-living/keyfree-smart-lock/

**Comment by pdlove** [comment]
As long as smartthings works with it, it will do what you want. I can say "lock all my doors" and "are the doors locked".
I also have it in a scene so I can say "Approaching the Garage". To open my garage, unlock the door and turn on the laundry room night...

**Comment by simonrb2000** [comment]
Brilliant, thank you. Yes it says it works with Smartthings. 

Thanks again

**Comment by h00pak** [comment]
I have a Yale lock connected to my Smartthings phone app but I just get this when I start home bridge:

[11/16/2016, 8:26:56 PM] [SmartThings] Initializing homebridge-smartthings.SmartThings platform...
[11/16/2016, 8:26:56 PM] [SmartThings] Fetching Smart Things devices.
[11/16/2016, 8:26:56 PM] [SmartThings] Error received type undefined - undefined
[11/16/2016, 8:26:56 PM] [SmartThings] Unknown Capabilities: []

**Comment by h00pak** [comment]
Hi, I have a Yale Real Living deadbolt. It works perfectly through the HomeApp and from both the iOS control center and the Smartthings app. But when I try to use Siri, I just get, "I didn't hear back."
Here is the initialization sequence from Homebridge - any ideas? Thank you.
Polling Seconds are 600 and Update Seconds are 1.
[11/17/2016, 8:22:07 PM] [SmartThings] Initializing SmartThings platform...
[11/17/2016, 8:22:07 PM] [SmartThings] Fetching Smart Things devices.
[11/17/2016, 8:22:07 PM] [SmartThings] Device Added - Group locks, Name Front Door, ID 7f36f0d2-94c1-4b0c-9072-c24488eb6e57
[11/17/2016, 8:22:07 PM] [SmartThings] Unknown Capabilities: []
[11/17/2016, 8:22:07 PM] [SmartThings] Initializing platform accessory 'Front Door'...

**Comment by whitej10** [comment]
I've bought the Yale Lock and it works brilliantly with the smart things hub. I can't get it to appear in the Home app or access it through HomeKit. I've setup up the json as per instructions (copied from the app). Any suggestions?

**Comment by whitej10** [comment]
I also have h00paks issue. Any suggestions?

**Comment by h00pak** [comment]
Here’s what my entry in config.json looks like for my smartthing hub. I’m guessing that you’re also using Homebridge to connect everything.

{
	"bridge": {
		"name": "Homebridge",
		"username": “AC:22:EE:Q1:MN:20",
		"port": 12345,
		"pin": "123-45-678"
	},

	"platforms": [

	{
			"platform": "homebridge-smartthings.SmartThings",
			"name": "SmartThings",
			"app_url": "https://graph-na02-useast1.api.smartthings.com:587/api/smartapps/installations/",
			"app_id": "2fd98e6j-4c50-4d8e-bf86-cada2d78a719",
			"access_token": "604a14d1-bc6a-4b1f-pokd-2a7n7829e5f7",
			"polling_seconds": 60,
			"update_seconds": 1
		}
	
	]
}

**Comment by pdlove** [comment]
@h00pak and @whitej10  Is Siri being more responsive with version 0.5.1 of the plugin?

**Comment by h00pak** [comment]
Unfortunately, I'm getting rate limit errors and can't seem to get them to stop. So I can't answer your question.

**Comment by h00pak** [comment]
Okay, I've updated the json app in my smartthings account and seem to be back where I was. Siri is still responsive to all the zwave devices except my Yale lock. It still will not respond to Siri. I just get a message back that Siri didn't hear back from my devices. 
But I'm no worse off.

**Comment by pdlove** [comment]
As of IOS version 10 Siri only acknowledges the lock was triggered when the new state actually updates as locked.
When you tap it in Home, how long does it take for it to stop spinning and acknowledge it is locked or unlocked?

**Comment by h00pak** [comment]
It really seems to vary. Sometimes it might be 30 seconds. Sometimes, ten. I have another lock that is connected via HomeKit and the speed of tapping the icon is comparable to the one connected via homebridge. The only difference is that Siri commands work fine for it for me. I grabbed this screen shot a few minutes ago, and it's the homebridge-connected lock.  The icon shows unknown state when I tapped it to unlock but the notification shows Unlocked.
